src/internal/incompatibility.rs

```diff
@@ -52,19 +52,16 @@ enum Kind<P: Package, V: Version> {
     DerivedFrom(IncompId<P, V>, IncompId<P, V>),
 }
 
-/// A type alias for a pair of [Package] and a corresponding [Term].
-pub type PackageTerm<P, V> = (P, Term<V>);
-
 /// A Relation describes how a set of terms can be compared to an incompatibility.
 /// Typically, the set of terms comes from the partial solution.
 #[derive(Eq, PartialEq)]
-pub enum Relation<P: Package, V: Version> {
+pub enum Relation<P: Package> {
     /// We say that a set of terms S satisfies an incompatibility I
     /// if S satisfies every term in I.
     Satisfied,
     /// We say that S contradicts I
     /// if S contradicts at least one term in I.
-    Contradicted(PackageTerm<P, V>),
+    Contradicted(P),
```

Comment:
This change is unrelated right? I suppose it's fine if we actually don't need the `Term` in that variant. We might want to verify though that it isn't the source of the perf improvement instead of the hash map changes. Seems very unlikely but things are sometimes surprising.

Reply:
Oops yes this is unrelated. I can back it out if you don't want it, or don't want it in this PR. I have bean running all tests using this commit as a baseline, just incase the removal of the one `Range<V>.clone()` mattered. But it does not make a perf difference from my measurements.

Reply:
alright we can keep the change in then